near-sdk-macros/src/core_impl/event/mod.rs

```diff
@@ -0,0 +1,75 @@
+use proc_macro::TokenStream;
+
+use proc_macro2::Span;
+use quote::quote;
+use syn::{parse_quote, ItemEnum};
+
+/// This attribute macro is used on a enum and its implementations
+/// to generate the necessary code to format standard event logs.
+///
+/// This macro will generate code to load and deserialize state if the `self` parameter is included
+/// as well as saving it back to state if `&mut self` is used.
+///
+/// For parameter serialization, this macro will generate a struct with all of the parameters as
+/// fields and derive deserialization for it. By default this will be JSON deserialized with `serde`
+/// but can be overwritten by using `#[serializer(borsh)]`.
+///
+/// `#[near_bindgen]` will also handle serializing and setting the return value of the
+/// function execution based on what type is returned by the function. By default, this will be
+/// done through `serde` serialized as JSON, but this can be overwritten using
+/// `#[result_serializer(borsh)]`.
+///
+/// # Examples
+///
+/// ```ignore
+/// use near_sdk::{near_events, Event};
+///
+/// #[near_events]
```

Comment:
I feel I might be missing something, why do we need two ways of achieving the same thing? I mean there is `#[near_bindgen(events)]` and this `#[near_events]`.

Reply:
I will change this. I made the conversion from `#[near_events]` to `#[near_bindgen(events)]` yesterday.